typescript/ws-protocol/src/FoxgloveServer.ts

```diff
@@ -599,4 +609,39 @@ export default class FoxgloveServer {
 
     connection.send(msg);
   }
+
+  /**
+   * Send a response to a fetchAsset request
+   * @param response The response to send
+   * @param connection Connection of the client that called the service
+   */
+  sendFetchAssetResponse(response: FetchAssetResponse, connection: IWebSocket): void {
+    const isSuccess = response.status === FetchAssetStatus.SUCCESS;
+    const mediaTypeOrErrMsg = textEncoder.encode(
+      isSuccess ? response.mediaType : response.errorMsg,
+    );
+    const dataLength = isSuccess ? response.data.byteLength : 0;
+    const payload = new Uint8Array(1 + 4 + 1 + 4 + mediaTypeOrErrMsg.length + dataLength);
+    const view = new DataView(payload.buffer, payload.byteOffset, payload.byteLength);
+    let offset = 0;
+    view.setUint8(offset, BinaryOpcode.ASSET);
+    offset += 1;
+    view.setUint32(offset, response.requestId, true);
+    offset += 4;
+    view.setUint8(offset, response.status);
+    offset += 1;
+    view.setUint32(offset, mediaTypeOrErrMsg.length, true);
+    offset += 4;
+    payload.set(mediaTypeOrErrMsg, offset);
+    offset += mediaTypeOrErrMsg.length;
+
+    if (isSuccess) {
+      payload.set(
+        new Uint8Array(response.data.buffer, response.data.byteOffset, response.data.byteLength),
```

Comment:
why not just `response.data`?

Reply:
`response.data` is a `DataView`, so it gives

```
Argument of type 'DataView' is not assignable to parameter of type 'ArrayLike<number>'
```

I guess that I should have used `Uint8Array` in the type definition, but I'm using `DataView` also for other types, so at least it's consistent.